pkg/signup/signup_service.go

```diff
@@ -0,0 +1,105 @@
+package signup
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	crtapi "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/registration-service/pkg/kubeclient"
+
+	errors2 "github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/client-go/rest"
+)
+
+type SignupServiceConfiguration interface {
+	GetNamespace() string
+}
+
+type SignupService interface {
+	CreateUserSignup(ctx context.Context, username, userID string) (*crtapi.UserSignup, error)
+}
+
+type SignupServiceImpl struct {
```

Comment:
I know that previously (in fabric8) we followed that convention: 
Interfaces: `type SomeName interface {}`
Implementations: `type SomeNameImpl struct {}`

But looking at kubernetes I noticed that they follow a different convention:
Interfaces: `type SomeNameInterface interface {}`
Implementations: `type SomeName struct {}`

We followed that kube convention in our kube client too (`UserSignup` and `UserSignupInterface` for example).

I would stick with the kube one.

Reply:
I know that I agreed it was better to be consistent with naming, but having thought about this some more I don't think it's necessarily better to conform to the Kubernetes naming standard for things that are actually outside of the Kubernetes framework.  So for the actual kube clients that we have created, I agree that they should conform to Kubernetes naming standards, however  `SignupService` is not a Kubernetes object, it's a service within our registration web app; while it talks to Kubernetes, it itself is designed as a standalone, "traditional" web application and as I mentioned previously I have more of an academic objection to the Kubernetes naming convention which I would rather not propagate further than we need to.

Reply:
OKey :)